packages/transport-commons/src/routing/index.ts

```diff
@@ -4,42 +4,45 @@ import { Router } from './router';
 declare module '@feathersjs/feathers/lib/declarations' {
   interface RouteLookup {
     service: Service,
-    params: { [key: string]: string }
+    params: { [key: string]: any }
   }
 
   interface Application<Services, Settings> {  // eslint-disable-line
-    routes: Router;
+    routes: Router<{
+      service: Service,
+      params?: { [key: string]: any }
+    }>;
     lookup (path: string): RouteLookup;
   }
 }
 
 export * from './router';
 
-export const routing = () => (app: Application) => {
-  if (typeof app.lookup === 'function') {
-    return;
+const lookup = function(this: Application, path: string) {
+  const result = this.routes.lookup(path);
+
+  if (result === null) {
+    return null;
   }
 
-  const routes = new Router();
+  const { params: colonParams, data: { service, params: dataParams } } = result;
 
-  Object.assign(app, {
-    routes,
-    lookup (this: Application, path: string) {
-      const result = this.routes.lookup(path);
+  const params = dataParams ? { ...dataParams, ...colonParams } : colonParams;
 
-      if (result !== null) {
-        const { params, data: service } = result;
+  return { service, params };
+};
 
-        return { params, service };
-      }
+export const routing = () => (app: Application) => {
+  if (typeof app.lookup === 'function') {
+    return;
+  }
 
-      return result;
-    }
-  });
+  app.routes = new Router();
+  app.lookup = lookup;
 
   // Add a mixin that registers a service on the router
   app.mixins.push((service: Service, path: string) => {
```

Comment:
Essentially this would become a

```ts
app.mixins.push((service: Service, path: string, options?: ServiceOptions) => {
  const { params = {} } = options;

  app.routes.insert(path, { service, params });
})
```

Reply:
Make sense, but wouldn't `route` be a better name for that option?

Reply:
Makes sense. But i think that `params` can be confused with `params` on context, meaning that it looks like a value will be merged directly into `params`, not into `params.route`. The clearest name is `routeParams`, but it is two words.

Reply:
Added `routeParams` to service options. Also fixed `ServiceMixin` type a little (`options` argument can't be undefined) and updated `route` on `params` to reflect that it can contain any values now, not only string ones.

Unresolve the conversation if you think that `params` is better name (I don't have strong opinion on this and ok with whatever).